File: xFormat/xFormatCommand.m
//
//  xFormatCommand.m
//  xFormat
//
//  Created by cyan on 16/6/19.
//  Copyright © 2016年 cyan. All rights reserved.
//

#import "xFormatCommand.h"
#import "xFormat.h"

@implementation xFormatCommand

- (NSDictionary *)handlers {
    static NSDictionary *_instance;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _instance = @{
            @"xformat.json": ^NSString *(NSString *text) { return FormatJSON(text); },
        };
    });
    return _instance;
}

@end

File: xFormat/xFormatCommand.swift
//
//  ██╗  ██╗████████╗███████╗██╗  ██╗████████╗
//  ╚██╗██╔╝╚══██╔══╝██╔════╝╚██╗██╔╝╚══██╔══╝
//   ╚███╔╝    ██║   █████╗   ╚███╔╝    ██║
//   ██╔██╗    ██║   ██╔══╝   ██╔██╗    ██║
//  ██╔╝ ██╗   ██║   ███████╗██╔╝ ██╗   ██║
//  ╚═╝  ╚═╝   ╚═╝   ╚══════╝╚═╝  ╚═╝   ╚═╝
//
//  xFormatCommand.swift
//  xFormat (https://github.com/cyanzhong/xTextHandler/)
//
//  Created by cyan on 16/7/4.
//  Copyright © 2016年 cyan. All rights reserved.
//

import Foundation

class xFormatCommand: xTextCommand {
    override func handlers() -> Dictionary<String, xTextModifyHandler> {
        return [
            "xformat.json": { text -> String in vkBeautify(string: text, type: "json") },
            "xformat.xml": { text -> String in vkBeautify(string: text, type: "xml") },
            "xformat.css": { text -> String in vkBeautify(string: text, type: "css") },
            "xformat.sql": { text -> String in vkBeautify(string: text, type: "sql") },
        ]
    }
}
